File: web/webapp/src/app/domain/model/domain-id.ts
import {ContextId} from './context-id';

export class DomainId {

  private static IDS = new Map<string, DomainId>();

  private constructor(public readonly contextId: ContextId,
                      public readonly  name: string) {
  }

  static fromNames(context: string, domain: string) {
    const key = `${context}/${domain}`;
    if (DomainId.IDS.has(key)) {
      return DomainId.IDS.get(key);
    }
    const id = new DomainId(ContextId.fromName(context), domain);
    DomainId.IDS.set(key, id);
    return id;
  }
}

File: discovery-domain/src/main/kotlin/io/landzcape/domain/DomainId.kt
package io.landzcape.domain

data class DomainId(val name: String, val context: ContextId) {

    companion object {
        fun from(name: String?, context: ContextId?): DomainId? {
            if(name!=null && context != null) {
                return DomainId(name, context)
            }
            return null
        }
    }
}